crates/bevy_time/src/lib.rs

```diff
@@ -70,21 +77,27 @@ pub fn create_time_channels() -> (TimeSender, TimeReceiver) {
 /// there to this system through channels. Otherwise the time is updated in this system.
 fn time_system(
     mut time: ResMut<Time>,
-    wanted_time: Option<Res<DesiredTime>>,
+    update_strategy: Res<TimeUpdateStrategy>,
     time_recv: Option<Res<TimeReceiver>>,
     mut has_received_time: Local<bool>,
 ) {
-    if let Some(time_recv) = time_recv {
-        // TODO: Figure out how to handle this when using pipelined rendering.
-        if let Ok(new_time) = time_recv.0.try_recv() {
-            time.update_with_instant(new_time);
-            *has_received_time = true;
-        } else if *has_received_time {
-            warn!("time_system did not receive the time from the render world! Calculations depending on the time may be incorrect.");
+    match update_strategy.as_ref() {
+        TimeUpdateStrategy::Automatic => {
+            if let Some(time_recv) = time_recv {
+                // TODO: Figure out how to handle this when using pipelined rendering.
+                if let Ok(new_time) = time_recv.0.try_recv() {
+                    time.update_with_instant(new_time);
+                    *has_received_time = true;
+                } else if *has_received_time {
+                    warn!("time_system did not receive the time from the render world! Calculations depending on the time may be incorrect.");
+                }
+            } else {
+                time.update();
+            }
+        }
+        TimeUpdateStrategy::ManualInstant(instant) => time.update_with_instant(*instant),
```

Comment:
You probably want to try_recv on the channel in the manual cases too and discard the value. The time channel only can have 2 times in it and will panic if the channel gets full.